### ISSUE TITLE
LineChart point selection issue

### ISSUE THREAD

**Issue by ncubelic** [danielgindi/Charts#5061]
> ℹ Please fill out this template when filing an issue.
> All lines beginning with an ℹ symbol instruct you with what info we expect. You can delete those lines once you've filled in the info.
>
> Per our [*CONTRIBUTING guidelines](https://github.com/jjatie/Charts/blob/master/CONTRIBUTING.md), we use GitHub for
> bugs and feature requests, not general support. Other issues should be opened on Stack Overflow with the tag [`ios-charts`.](https://stackoverflow.com/questions/tagged/ios-charts)
>
> Please remove this line and everything above it before submitting.

* [ ] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/jjatie/Charts/blob/master/CONTRIBUTING.md).

## What did you do?

I'm having issues with LineChart when using numbers on y axis which differs on second or third decimal place. I can't select point on a graph. No matter which point on graph is selected, only first or last point is selected.

## What did you expect to happen?

Point which is selected should be highlighted and selected. 

## What happened instead?

First or last point is selected.

## Charts Environment

**Charts version/Branch/Commit Number:**
4.1.0

**Xcode version:**
Xcode 14.3

**Swift version:**
Swift 5

**Platform(s) running Charts:**
iOS

**macOS version running Xcode:**

## Demo Project

ℹ Please link to or upload a project we can download that reproduces the issue.
[LineChart.zip](https://github.com/danielgindi/Charts/files/11542399/LineChart.zip)

**Comment by stas44812** [comment]
did you resolve this? facing same problem

**Comment by ncubelic** [comment]
@stas44812 Yes I resolved the issue. Problem was due to unsorted data set. I don’t know logic behind, but when you pass sorted data set (in my case ba date), everything works perfectly.

**Comment by stas44812** [comment]
thanks man! will try
